R/ResultData.R

```diff
@@ -68,6 +70,12 @@ ResultData = R6Class("ResultData",
           set(data, j = "resampling", value = NULL)
           set(data, j = "param_values", value = NULL)
 
+          # add extra data to fact table
+          if (!is.null(data_extra)) {
+            assert_list(data_extra, len = nrow(data))
+            set(data, j = "data_extra", value = data_extra)
```

Comment:
list cols can be tricky in data.table, in particular when there are 0 or 1 row tables involved. See

* https://github.com/Rdatatable/data.table/issues/4568
* https://github.com/Rdatatable/data.table/issues/6104
* https://github.com/Rdatatable/data.table/issues/4569

Idk how relevant this is here. E.g. if all tests involve having tables here that have strictly more than 1 row, consider also having a test has  a single row here. But ignore this comment if you know this is not an issue.

Reply:
Thanks for pointing this out. Should be a always a list column now.